mlir/include/mlir/Analysis/Presburger/Fraction.h

```diff
@@ -124,6 +129,14 @@ inline Fraction operator-(const Fraction &x, const Fraction &y) {
   return reduce(Fraction(x.num * y.den - x.den * y.num, x.den * y.den));
 }
 
+// Find the integer nearest to a given fraction.
+inline MPInt round(const Fraction &f) {
+  MPInt rem = f.num % f.den;
+  if (rem < Fraction(f.den, 2))
+    return (f.num - rem) / f.den;
+  return (f.num + f.den - rem) / f.den;
```

Comment:
Sorry I just realized this can be further simplified

`return f.num/f.den + (f.num > f.den/2);`